build.sbt

```diff
@@ -133,8 +141,7 @@ lazy val sdkSpring = project
     buildInfoPackage := "kalix.springsdk",
     Compile / akkaGrpcGeneratedSources := Seq(AkkaGrpc.Server),
     Compile / akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Scala),
-    Test / javacOptions ++= Seq("-parameters", "--release", "17"), // for Jackson
-    Test / scalacOptions ++= Seq("-release", "17"),
```

Comment:
I guess these were misplaced in Test scope

Reply:
Renato had the same thought 2 days ago but then said "We only want the tests to be on 17, not the production artifacts." [here](https://github.com/lightbend/kalix-jvm-sdk/pull/1427#discussion_r1092041066). I'm not sure what's the rational though.

Reply:
I changed it and then reverted in the PR for the Instant. Now, if SpringBoot 3.0 requires 17, then we can simply say that Spring SDK requires 17. 

For us that will be perfect, because we will be able to use records in our code base.